src/ray/gcs/gcs_client/test/service_based_gcs_client_test.cc

```diff
@@ -1143,6 +1143,45 @@ TEST_F(ServiceBasedGcsClientTest, TestGcsRedisFailureDetector) {
   RAY_CHECK(gcs_server_->IsStopped());
 }
 
+TEST_F(ServiceBasedGcsClientTest, TestMultiThreadSubAndUnsub) {
+  auto sub_finished_count = std::make_shared<std::atomic<int>>(0);
+  int size = 5;
+  std::vector<std::unique_ptr<std::thread>> threads;
+  threads.resize(size);
+
+  // Multithreading subscribe/resubscribe/unsubscribe actors.
+  auto job_id = JobID::FromInt(1);
+  for (int index = 0; index < size; ++index) {
+    threads[index].reset(new std::thread([this, job_id] {
+      auto actor_id = ActorID::Of(job_id, RandomTaskId(), 0);
```

Comment:
Can we run loop inside here like 10~20 times?

Reply:
fixed